src/filing/actions/receiveEdit.js

```diff
@@ -1,14 +1,66 @@
+import React from 'react'
 import * as types from '../constants'
 
+const PATTERN_1111 = /1111/
+
+/**
+ *  Restrict the displayed fields to those whose value matches a pattern 
+ */
+const filterEditFields = (rows = [], pattern = PATTERN_1111) => {
+  let fieldMatches = {}
+
+  // Identify field values that match this pattern
+  rows.forEach((r) => {
+    if (!r.fields) return
+    r.fields.forEach((f) => {
+      if (!f.value.match(pattern)) return
+      fieldMatches[f.name] = true
+    })
+  })
+
+  // Retain only those fields
+  rows.forEach((r) => {
+    if (!r.fields) return
+    r.fields = r.fields.filter((f) => {
+      if (!fieldMatches[f.name]) return null
+      if (f.value.match(pattern)) {
+        // Highlight the offending field
+        f.value = <span className='highlight'>{f.value}</span>
+      }
+      return f
+    })
+  })
+}
```

Comment:
I think generally we'd like to avoid side-effect style of coding if we can; so rather than modifying the argument passed in, I think returning a result would be cleaner; also I believe 1 iteration should be able to achieve this, instead of looping through the collection twice; think we can do `rows.map(...)` and return that result.